packages/node_modules/@ciscospark/internal-plugin-mercury/src/mercury.js

```diff
@@ -193,6 +195,12 @@ const Mercury = SparkPlugin.extend({
           this.backoffCall.abort();
           return callback(reason);
         }
+        else if (reason instanceof ConnectionError) {
+          this.logger.info(`mercury: received a generic connection error, will try to connect to another datacenter`);
+          return this.spark.internal.device.markUrlFailedAndGetNew(`mercuryConnection`)
```

Comment:
maybe instead of marking a service failed, we should mark a url as failed and let the code in wdm take care of figuring out the service.